### PR TITLE
chore: use pnpm as package manager

### PR DESCRIPTION
**Pull request by bjoerge**
### Description
This migrates this repo to use pnpm instead of (legacy) yarn 1.x

### Testing
- remove all node_modules etc.
- run pnpm install
- verify that things still works (test studios, scripts, etc)

### Notes for release

n/a - not user facing

### CODE REVIEW

**Comment by bjoerge** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5533?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `next`
  * **PR #5531** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5531?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5534** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5534?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5532** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5532?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5533** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5533?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/sanity-io/sanity/5533?utm_source=stack-comment).

File: .github/workflows/docReport.yml [hunk 1]
```diff
@@ -18,25 +18,32 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/docReport.yml:43]:
```suggestion
        run: pnpm install
```

File: .github/workflows/e2e-ct.yml [hunk 2]
```diff
@@ -17,22 +17,26 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --immutable --immutable-cache --check-cache
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/e2e-ct.yml:39]:
```suggestion
        run: pnpm install
```

File: .github/workflows/e2e-pte.yml [hunk 3]
```diff
@@ -25,22 +25,25 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/e2e-pte.yml:46]:
```suggestion
        run: pnpm install
```

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -23,22 +23,25 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/e2e.yml:44]:
```suggestion
        run: pnpm install
```

File: .github/workflows/e2e.yml [hunk 5]
```diff
@@ -114,22 +117,25 @@ jobs:
...
```
Comment (stipsan) [inline .github/workflows/e2e.yml:138]:
```suggestion
        run: pnpm install
```

File: .github/workflows/e2e.yml [hunk 6]
```diff
@@ -203,22 +209,25 @@ jobs:
...
```
Comment (stipsan) [inline .github/workflows/e2e.yml:230]:
```suggestion
        run: pnpm install
```

File: .github/workflows/etl.yml [hunk 7]
```diff
@@ -18,25 +18,32 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/etl.yml:43]:
```suggestion
        run: pnpm install
```

File: .github/workflows/formatCheck.yml [hunk 8]
```diff
@@ -16,22 +16,25 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/formatCheck.yml:37]:
```suggestion
        run: pnpm install
```

File: .github/workflows/lint-fix-if-needed.yml [hunk 9]
```diff
@@ -24,39 +24,43 @@ jobs:
...
-          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/lint-fix-if-needed.yml:45]:
```suggestion
        run: pnpm install
```

File: .github/workflows/lint-fix-if-needed.yml [hunk 10]
```diff
@@ -24,39 +24,43 @@ jobs:
...
         with:
           app_id: ${{ secrets.ECOSPARK_APP_ID }}
           private_key: ${{ secrets.ECOSPARK_APP_PRIVATE_KEY }}
       - uses: peter-evans/create-pull-request@b1ddad2c994a25fbc81a28b3ec0e368bb2021c50 # v6
-        # Run even if `yarn lint:fix` fails
+        # Run even if `lint:fix` fails
         if: always()
         with:
-          body: "I ran `yarn lint:fix --rule 'prettier/prettier: [off]'` 🧑‍💻"
+          body: "I ran `lint:fix --rule 'prettier/prettier: [off]'` 🧑‍💻"
```
Comment (stipsan) [inline .github/workflows/lint-fix-if-needed.yml:63]:
```suggestion
          body: "I ran `pnpm lint:fix --rule 'prettier/prettier: [off]'` 🧑‍💻"
```

File: .github/workflows/lintPr.yml [hunk 11]
```diff
@@ -18,22 +18,25 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/lintPr.yml:39]:
```suggestion
        run: pnpm install
```

File: .github/workflows/perf.yml [hunk 12]
```diff
@@ -31,7 +34,7 @@ jobs:
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
         working-directory: ./perf
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/perf.yml:37]:
```suggestion
        run: pnpm install
```

File: .github/workflows/pnpm-if-needed.yml [hunk 13]
```diff
@@ -17,38 +17,46 @@ permissions:
...
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --ignore-scripts --frozen-lockfile --update-checksums --force
-      - run: npx yarn-deduplicate -- yarn.lock --strategy highest
+        run: pnpm install --ignore-scripts --frozen-lockfile --force
+      - run: pnpm dedupe
```
Comment (stipsan) [inline .github/workflows/pnpm-if-needed.yml:49]:
```suggestion
        run: pnpm install --config.ignore-scripts=true
      - run: pnpm dedupe --config.ignore-scripts=true
```

File: .github/workflows/pr-cleanup.yml [hunk 14]
```diff
@@ -16,28 +16,35 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/pr-cleanup.yml:41]:
```suggestion
        run: pnpm install
```

File: .github/workflows/pr-cleanup.yml [hunk 15]
```diff
@@ -59,19 +66,19 @@ jobs:
...
```
Comment (stipsan) [inline .github/workflows/pr-cleanup.yml:77]:
```suggestion
        run: pnpm install
```

File: .github/workflows/typeCheck.yml [hunk 16]
```diff
@@ -11,27 +11,30 @@ jobs:
...
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/typeCheck.yml:32]:
```suggestion
        run: pnpm install
```

File: .github/workflows/test.yml [hunk 17]
```diff
@@ -126,14 +132,14 @@ jobs:
 
       - name: Install project dependencies
         if: steps.restore-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/test.yml:135]:
```suggestion
        run: pnpm install
```

File: .github/workflows/test.yml [hunk 18]
```diff
@@ -62,10 +65,10 @@ jobs:
 
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile
+        run: pnpm install --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/test.yml:68]:
```suggestion
        run: pnpm install
```

File: .github/workflows/prettier-if-needed.yml [hunk 19]
```diff
@@ -24,37 +24,41 @@ jobs:
...
+        run: git restore .github/workflows pnpm-lock.yaml
       - uses: tibdex/github-app-token@3beb63f4bd073e61482598c45c71c1019b59b73a # v2
         id: generate-token
         with:
           app_id: ${{ secrets.ECOSPARK_APP_ID }}
           private_key: ${{ secrets.ECOSPARK_APP_PRIVATE_KEY }}
       - uses: peter-evans/create-pull-request@b1ddad2c994a25fbc81a28b3ec0e368bb2021c50 # v6
         with:
-          body: I ran `yarn prettier` 🧑‍💻
+          body: I ran `prettier` 🧑‍💻
```
Comment (stipsan) [inline .github/workflows/prettier-if-needed.yml:61]:
```suggestion
          body: I ran `pnpm prettier` 🧑‍💻
```

File: .github/workflows/prettier-if-needed.yml [hunk 20]
```diff
@@ -24,37 +24,41 @@ jobs:
...
-          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/yarn.lock') }}
+          key: ${{ runner.os }}-modules-${{ env.cache-name }}-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-modules-${{ env.cache-name }}-
             ${{ runner.os }}-modules-
             ${{ runner.os }}-
       - name: Install project dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: yarn install --ignore-scripts --frozen-lockfile
+        run: pnpm install --ignore-scripts --frozen-lockfile
```
Comment (stipsan) [inline .github/workflows/prettier-if-needed.yml:45]:
```suggestion
        run: pnpm install --config.ignore-scripts=true
```

**Review by stipsan — APPROVED** [review]
Looks good! Added some optional suggestions 🥰

**Review by binoy14 — CHANGES_REQUESTED** [review]
just one change requested for handling node_modules caching besides that it looks good

**Review by binoy14 — APPROVED** [review]

**Pull request merged**